pkg/server/endpoints/node/handler.go

```diff
@@ -234,8 +235,16 @@ func (h *Handler) FetchX509SVID(server node.Node_FetchX509SVIDServer) (err error
 			return err
 		}
 
-		for spiffeID := range svids {
+		for spiffeID, svid := range svids {
 			counter.AddLabel("spiffe_id", spiffeID)
+
+			ttl := time.Until(time.Unix(svid.ExpiresAt, 0))
+			metrics.SetGaugeWithLabels(
+				[]string{"node_api", "x509_svid", "fetch", "ttl"},
+				float32(ttl.Seconds()),
+				[]telemetry.Label{
+					{Name: "spiffe_id", Value: spiffeID},
```

Comment:
I think that in practice, measuring this from node api handler will be less useful. First, we already know what the ttl should be because it is set in the registration entry as a matter of policy. Second, I think this might get confusing because it is possible to have more than one registration entry for the same SPIFFE ID, which might have differing TTLs.. in this case, the metric points will be indistinguishable, some backends will actually take the average value for display purposes.

I recommend removing this point and emitting only from workload api

Reply:
Makes sense, will remove this. Thank you!